variable/creation.cpp

```diff
@@ -11,18 +11,29 @@
 namespace scipp::variable {
 
 Variable empty(const Dimensions &dims, const units::Unit &unit,
-               const DType type, const bool variances) {
-  return variableFactory().create(type, dims, unit, variances);
+               const DType type, const bool with_variances) {
+  return variableFactory().create(type, dims, unit, with_variances);
 }
 
-Variable ones(const Dimensions &dims, const units::Unit &unit, const DType type,
-              const bool variances) {
+Variable full(const Dimensions &dims, const int value, const units::Unit &unit,
+              const DType type, const std::optional<int> variance) {
   const auto prototype =
-      variances ? Variable{type, Dimensions{}, unit, Values{1}, Variances{1}}
-                : Variable{type, Dimensions{}, unit, Values{1}};
+      variance.has_value() ? Variable{type, Dimensions{}, unit, Values{value},
+                                      Variances{*variance}}
+                           : Variable{type, Dimensions{}, unit, Values{value}};
   return copy(broadcast(prototype, dims));
 }
 
+Variable zeros(const Dimensions &dims, const units::Unit &unit,
+               const DType type, const bool with_variances) {
+  return full(dims, 0, unit, type, with_variances ? 0 : std::optional<int>{});
```

Comment:
Have you checked if this has the same performance as the `Variable` constructor, which zeros by default?

Reply:
Which one are you talking about? `makeVariable` is not easily accessible here and `variableFactory().create` does not initialise the elements.

Reply:
I suppose this is something that used `callDType` with `makeVariable` in the old constructor code?

Reply:
I mean yes, the one based on `makeVariable`.